tiledb/api/src/array/fragment_info.rs

```diff
@@ -0,0 +1,879 @@
+                self.context.capi(),
+                self.capi(),
+                frag_idx,
+                &mut c_schema,
+            )
+        })?;
+
+        Ok(Schema::new(self.context, RawSchema::Owned(c_schema)))
+    }
+
+    pub fn schema_name(&self, frag_idx: u32) -> TileDBResult<String> {
+        let mut c_str: *const std::ffi::c_char = out_ptr!();
+        self.capi_return(unsafe {
+            ffi::tiledb_fragment_info_get_array_schema_name(
+                self.context.capi(),
+                self.capi(),
+                frag_idx,
+                &mut c_str,
+            )
+        })?;
+
+        // N.B. This API only lends a pointer to an internally managed
+        // std::string, thus we do *not* want to free it.
+        let name = unsafe { std::ffi::CStr::from_ptr(c_str) };
+        Ok(String::from(name.to_string_lossy()))
+    }
+
+    pub fn fragment_name(&self, frag_idx: u32) -> TileDBResult<String> {
+        let mut c_str: *mut ffi::tiledb_string_t = out_ptr!();
+        self.capi_return(unsafe {
+            ffi::tiledb_fragment_info_get_fragment_name_v2(
+                self.context.capi(),
+                self.capi(),
+                frag_idx,
+                &mut c_str,
+            )
+        })?;
+
+        let str = TDBString::from_raw(RawTDBString::Owned(c_str));
+        str.to_string()
+    }
+
+    pub fn fragment_uri(&self, frag_idx: u32) -> TileDBResult<String> {
+        let mut c_str: *const std::ffi::c_char = out_ptr!();
+        self.capi_return(unsafe {
+            ffi::tiledb_fragment_info_get_fragment_uri(
+                self.context.capi(),
+                self.capi(),
+                frag_idx,
+                &mut c_str,
+            )
+        })?;
+
+        // N.B. This API only lends a pointer to an internally managed
+        // std::string, thus we do *not* want to free it.
+        let name = unsafe { std::ffi::CStr::from_ptr(c_str) };
+        Ok(String::from(name.to_string_lossy()))
+    }
+
+    pub fn fragment_size(&self, frag_idx: u32) -> TileDBResult<u64> {
+        let mut size: u64 = 0;
+        self.capi_return(unsafe {
+            ffi::tiledb_fragment_info_get_fragment_size(
+                self.context.capi(),
+                self.capi(),
+                frag_idx,
+                &mut size,
+            )
+        })?;
+
+        Ok(size)
+    }
+
+    pub fn fragment_type(&self, frag_idx: u32) -> TileDBResult<FragmentType> {
+        let mut dense: i32 = 0;
+        self.capi_return(unsafe {
+            ffi::tiledb_fragment_info_get_dense(
+                self.context.capi(),
+                self.capi(),
+                frag_idx,
+                &mut dense,
+            )
+        })?;
+
+        if dense == 1 {
+            Ok(FragmentType::Dense)
+        } else {
+            Ok(FragmentType::Sparse)
+        }
+    }
+
+    pub fn timestamp_range(&self, frag_idx: u32) -> TileDBResult<(u64, u64)> {
```

Comment:
For the return type - for things like extents we have done `[T; 2]`. THis would be another item to make consistent.
At least as far as JSON is concerned the slice approach gives you something a little nicer, since you would have
`{ "timestamp_range": [start, end] }`
instead of
`{ "timestamp_range": { "0": start, "1": end }}` (which is what I assume the tuple type turns into)

The slice approach also lets you do things like iterate but I'm not really sure why one would want to do that

Reply:
Good call. Fixed.